cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -162,6 +162,9 @@ type DNS struct {
 
 	// ImageMeta allows to customize the image used for the DNS component
 	ImageMeta `json:",inline"`
+
+	// NodeLocalDNSCacheIP specifies the local IP to bind the DNS cache to. Defaults to "169.254.20.10"
+	NodeLocalDNSCacheIP string `json:"nodeLocalDNSCacheIP"`
```

Comment:
we figure out if `k8s-dns-node-cache` is going to receive good maintenance in the future. it's part of the dns repo and kube-dns is being deprecated eventually.

Reply:
Also we need to figure out what IP to use, ref: https://github.com/kubernetes/enhancements/blob/master/keps/sig-network/0030-nodelocal-dns-cache.md#choice-of-caching-agent

> The caching agent daemonset runs in hostNetwork mode in kube-system namespace with a Priority Class of “system-node-critical”. It listens for dns requests on a dummy interface created on the host. A separate ip address is assigned to this dummy interface, so that requests to kube-dns or any other custom service are not incorrectly intercepted by the caching agent. This will be a link-local ip address selected by the user. Each cluster node will have this dummy interface. This ip address will be passed on to kubelet via the --cluster-dns flag, if the feature is enabled.
> The selected link-local IP will be handled specially because of the NOTRACK rules described in the section below.

Also we need to make sure this is the long-term solution for minimum viable k8s clusters.

Reply:
> since it does nothing to the deployment of the main DNS addons.

I'm not sure about this feature actually **is becoming** the main DNS addon (together with the "normal" CoreDNS obviously, but...)